### ISSUE TITLE
"error: invalid application of ‘sizeof’ to incomplete type" in drivers/net/wireless/marvell/mwifiex/scan.c

### ISSUE THREAD

**Issue by GustavoARSilva** [KSPP/linux#51]
When replacing the one-element array **chan_scan_param[1]** with a flexible-array member in struct **mwifiex_chan_scan_param_set**:

```
diff --git a/drivers/net/wireless/marvell/mwifiex/fw.h b/drivers/net/wireless/marvell/mwifiex/fw.h
index a415d73a73e6..6617dd34e07a 100644
--- a/drivers/net/wireless/marvell/mwifiex/fw.h
+++ b/drivers/net/wireless/marvell/mwifiex/fw.h
@@ -780,7 +780,7 @@ struct mwifiex_chan_scan_param_set {
 
 struct mwifiex_ie_types_chan_list_param_set {
        struct mwifiex_ie_types_header header;
-       struct mwifiex_chan_scan_param_set chan_scan_param[1];
+       struct mwifiex_chan_scan_param_set chan_scan_param[];
 } __packed;
 
 struct mwifiex_ie_types_rxba_sync {
```

the following **errors** show up:

```
  CC      drivers/net/wireless/marvell/mwifiex/scan.o
drivers/net/wireless/marvell/mwifiex/scan.c: In function ‘mwifiex_scan_channel_list’:
drivers/net/wireless/marvell/mwifiex/scan.c:694:17: error: invalid application of ‘sizeof’ to incomplete type ‘struct mwifiex_chan_scan_param_set[]’
  694 |           sizeof(chan_tlv_out->chan_scan_param));
      |                 ^
drivers/net/wireless/marvell/mwifiex/scan.c:699:19: error: invalid application of ‘sizeof’ to incomplete type ‘struct mwifiex_chan_scan_param_set[]’
  699 |             sizeof(
      |                   ^
drivers/net/wireless/marvell/mwifiex/scan.c: In function ‘mwifiex_cmd_802_11_bg_scan_config’:
drivers/net/wireless/marvell/mwifiex/scan.c:2405:19: error: invalid application of ‘sizeof’ to incomplete type ‘struct mwifiex_chan_scan_param_set[]’
 2405 |             sizeof(
      |                   ^
scripts/Makefile.build:272: recipe for target 'drivers/net/wireless/marvell/mwifiex/scan.o' failed
```

See [flexible-array conversions](https://github.com/KSPP/linux/issues/21) issue.

**Comment by kees** [comment]
Sent https://lore.kernel.org/lkml/20240206163501.work.158-kees@kernel.org

**Comment by GustavoARSilva** [comment]
> Sent https://lore.kernel.org/lkml/20240206163501.work.158-kees@kernel.org

mmh... I don't see the actual flex-array transformation in that patch.

**Comment by kees** [comment]
Fixed in commit 14ddc470ba22057f7734245a6a521d764f8a7fbe
